code/modules/antagonists/heretic/items/heretic_blades.dm

```diff
@@ -50,8 +50,18 @@
 	return .
 
 /obj/item/melee/sickly_blade/attack_self(mob/user)
+	if(HAS_TRAIT(user, TRAIT_ELDRITCH_ARENA_PARTICIPANT))
+		user.balloon_alert(user, "can't escape!")
+		to_chat(user, span_hypnophrase(span_big("Cowardly sheep will be slaughtered!")))
+		playsound(src, SFX_SHATTER, 70, TRUE)
+		var/obj/item/bodypart/to_remove = user.get_active_hand()
+		to_remove.dismember()
+		qdel(src)
```

Comment:
Ngl, this lacks any sort of safety against accidental inputs and it's very, very harsh. Skill issues aside, you should definitely add a variable that increases each time the proc is called while inside the arena, and with some callbacks that reset the variable back to 0 after a few seconds, make it so it only chops the arm off if the value reaches 5 or 4 or any arbitrary number greater than 3.